src/main/java/com/google/devtools/build/lib/bazel/rules/python/python_stub_template.txt

```diff
@@ -71,17 +71,27 @@ def CreatePythonPathEntries(python_imports, module_space):
 
 # Find the runfiles tree
 def FindModuleSpace():
-  stub_filename = os.path.abspath(sys.argv[0])
-  module_space = stub_filename + ('.exe' if IsWindows() else '') + '.runfiles'
-  if os.path.isdir(module_space):
-    return module_space
-
-  runfiles_pattern = "(.*\.runfiles)/.*"
-  if IsWindows():
-    runfiles_pattern = "(.*\.runfiles)\\.*"
-  matchobj = re.match(runfiles_pattern, os.path.abspath(sys.argv[0]))
-  if matchobj:
-    return matchobj.group(1)
+  stub_filename = sys.argv[0]
+  if not os.path.isabs(stub_filename):
+    stub_filename = os.path.join(os.getcwd(), stub_filename)
+
+  while True:
+    module_space = stub_filename + ('.exe' if IsWindows() else '') + '.runfiles'
+    if os.path.isdir(module_space):
+      return module_space
+
+    runfiles_pattern = '(.*\.runfiles)' + ('\\\\' if IsWindows() else '/') + '.*'
```

Comment:
In the first string, do you need a `r` prefix for the `\`? I believe without it the `\` will be consumed by the string literal, not by the regex matcher.
```
r'(.*\.runfiles)'
```

Reply:
Yes.  I changed to using an 'r' prefixes for both strings.